### ISSUE TITLE
Rewrite in a performant language e.g. Golang

### ISSUE THREAD

**Issue by devnoname120** [AdamWagner/stackline#111]
Stackline spikes in CPU whenever I switch windows. This has a significant battery impact which would be reduced by using a compiled language such as Golang.

I may look into it in the future but it's low-priority in my TODO list.

**Comment by prurph** [comment]
Isn't Hammerspoon being used to actually render the icons?

**Comment by es183923** [comment]
literally everything depends on hammerspoon. The timers, rendering the app icons/pills, clicks, being updated on certain events, logging, etc. Some of this can be removed, but mainly rendering the icons/pills and clicks require hammerspoon.

**Comment by devnoname120** [comment]
Yes it would be a complete rewrite from scratch. I'll use this project as inspiration for what it will look like but otherwise there won't be much in common.

**Comment by mizlan** [comment]
@devnoname120 have you worked on it?

**Comment by devnoname120** [comment]
Nope I haven't

**Comment by liam-mackie** [comment]
I've written an alternative in Swift. You can find it here: https://github.com/liam-mackie/stackline-swift 😄 
Happy to get feedback and contributions, I've built it to be pretty specific to my own requirements, and it's very hacky. That being said, it's handled by automatically adding events to yabai rather than polling, and in my testing, the energy impact is pretty small.

**Comment by devnoname120** [comment]
Liam’s rewrite is working really well and fast, I recommend everyone to try!
